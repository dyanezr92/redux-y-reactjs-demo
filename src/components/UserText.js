import React from 'react'
import PropTypes from 'prop-types'

const UserText = ({ text, author, id, onDeleteClick }) => {
  const onDeleteUserTextClick = event => {
    event.preventDefault();
    onDeleteClick(id);
  };

  return <blockquote>
    <input type="hidden" value={id}/>

    {text}
    <footer>{ author }</footer>
    <img alt="Delete button" onClick={ onDeleteUserTextClick } src="https://wecision.com/wecision/images/icons/closeIcon.png" />
  </blockquote>
};

UserText.propTypes = {
  text: PropTypes.string,
  author: PropTypes.string,
  id: PropTypes.number.isRequired,
  onDeleteClick: PropTypes.func.isRequired
}

export default UserText
