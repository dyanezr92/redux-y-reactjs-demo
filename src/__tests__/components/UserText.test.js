import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'; // Smart components

import UserText from '../../components/UserText';

describe('<UserText key={1} text="Awesome comment" author="Finn" id={1} onDeleteClick={console::log} />', () => {
  describe('render()', () => {
    test('renders the user text', () => {
      const wrapper = shallow(
        <UserText key={1} text="Awesome comment" author="Finn" id={1} onDeleteClick={console.log} />
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
