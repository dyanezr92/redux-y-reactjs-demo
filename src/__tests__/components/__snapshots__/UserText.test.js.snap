// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserText key={1} text="Awesome comment" author="Finn" id={1} onDeleteClick={console::log} /> render() renders the user text 1`] = `
<blockquote>
  <input
    type="hidden"
    value={1}
  />
  Awesome comment
  <footer>
    Finn
  </footer>
  <img
    alt="Delete button"
    onClick={[Function]}
    src="https://wecision.com/wecision/images/icons/closeIcon.png"
  />
</blockquote>
`;
